src/constraints.jl

```diff
@@ -1280,3 +1280,57 @@ function num_constraints(model::Model; count_variable_in_set_constraints::Bool)
     end
     return ret
 end
+
+"""
+    all_constraints(
+        model::Model;
+        include_variable_in_set_constraints::Bool,
+    )::Vector{ConstraintRef}
+
+Return a list of all constraints in `model`.
+
+If `include_variable_in_set_constraints == true`, then `VariableRef` constraints
+such as `VariableRef`-in-`Integer` are included. To return only the structural
+constraints (e.g., the rows in the constraint matrix of a linear program), pass
+`include_variable_in_set_constraints = false`.
+
+Note that this function is type-unstable because it returns an abstractly typed
+vector. 
```

Comment:
We should at least give a pointer to somewhere where we show how to do this efficiently

Reply:
Pointing to here:
https://github.com/jump-dev/JuMP.jl/pull/3002/files#diff-374232ee51673809a2a65674f659b6dda32ecec8854a91876b2ff2566e1f07d2R957
maybe?

Reply:
I just think this docstring is the wrong place for an extended note on performance and how to do this differently. If someone is writing on top of JuMP, then can profile their code, or already know about function barriers.